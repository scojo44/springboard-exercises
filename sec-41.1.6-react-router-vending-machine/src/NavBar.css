.NavBar {
  list-style-type: none;
  background-color: rgba(0,0,0,60%);
}
.NavBar li {
  display: inline-block;
  margin: .5rem;
}
.NavBar .active img {
  transform: scale(1.2);
  transition: transform .2s ease-in-out;
}
.NavBar img {
  height: 10vh;
  max-height: 100px;
}
