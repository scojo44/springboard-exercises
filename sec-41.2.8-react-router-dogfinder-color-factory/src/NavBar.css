.NavBar {
  display: flex;
  background-color: darkblue;
  padding: .5rem 0;
}
.NavBar a,
.NavBar a:visited,
.NavBar a:link {
  color: white;
  font-weight: bold;
  text-decoration: none;
  padding: 0 1rem;
}
.NavBar div {
  text-align: right;
  flex: 1;
}