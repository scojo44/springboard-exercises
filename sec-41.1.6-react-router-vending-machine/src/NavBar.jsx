import React from 'react'
import {NavLink} from 'react-router-dom'
import './NavBar.css'

function NavBar() {
  return (
    <menu className='NavBar'>
      <li><NavLink to="/"><img src="https://cdn-icons-png.freepik.com/256/12438/12438220.png?semt=ais_hybrid" alt="React Router Vending Machine Home" /></NavLink></li>
      <li><NavLink to="/pibbxtra"><img src="https://upload.wikimedia.org/wikipedia/commons/9/9a/Pibb_Xtra_Logo.jpg" alt="Pibb Xtra" /></NavLink></li>
      <li><NavLink to="/drpepper"><img src="https://pbs.twimg.com/profile_images/1695185929286340608/jMfIR0OJ_400x400.jpg" alt="Dr. Pepper" /></NavLink></li>
      <li><NavLink to="/rc"><img src="https://www.giannabaldino.com/wp-content/uploads/2019/07/Png.png" alt="RC Cola" /></NavLink></li>
      <li><NavLink to="/sunkist/orange"><img src="https://contenthandler-raleys.fieldera.com/prod/342451/1/0/0/30086914-Planogram-Left.jpg" alt="Sunkist Orange" /></NavLink></li>
      <li><NavLink to="/sunkist/grape"><img src="https://www.midwestbev.com/cdn/shop/products/SunG_x700.jpg?v=1574366037" alt="Sunkist Grape" /></NavLink></li>
      <li><NavLink to="/7up/cherry"><img src="https://www.midwestbev.com/cdn/shop/products/Cherry_7up_x700.jpg?v=1574106619" alt="Cherry 7up" /></NavLink></li>
      <li><NavLink to="/rootbeer/aw"><img src="https://www.midwestbev.com/cdn/shop/products/awrb_x700.jpg" alt="A&W Root Beer" /></NavLink></li>
      <li><NavLink to="/rootbeer/xxx"><img src="http://blog.gourmetrootbeer.com/wp-content/uploads/2024/04/triplexxxRestaurant_big.jpg" alt="Triple XXX Root Beer" /></NavLink></li>
    </menu>
  )
}

export default NavBar
