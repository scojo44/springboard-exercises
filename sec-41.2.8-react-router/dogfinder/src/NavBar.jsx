import React from "react"
import { NavLink } from "react-router-dom"
import './NavBar.css'

const NavBar = ({dogNames}) => {
  return (
    <nav className="NavBar">
      <NavLink to='/dogs'>Dog Finder</NavLink>
      <div>
        {dogNames.map(name => <NavLink to={'/dogs/' + name.toLowerCase()} key={name}>{name}</NavLink>)}
      </div>
    </nav>
  );
};

export default NavBar
