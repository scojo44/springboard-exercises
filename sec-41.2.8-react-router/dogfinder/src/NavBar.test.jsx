import {render} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import {dogs} from './dogs.json'
import NavBar from './NavBar'

describe('NavBar Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <NavBar dogNames={dogs.map(d => d.name)} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <NavBar dogNames={dogs.map(d => d.name)} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
