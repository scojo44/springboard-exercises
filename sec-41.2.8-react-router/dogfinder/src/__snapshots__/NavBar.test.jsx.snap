// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      class=""
      href="/dogs"
    >
      Dog Finder
    </a>
    <div>
      <a
        class=""
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
      <a
        class=""
        href="/dogs/duke"
      >
        Duke
      </a>
      <a
        class=""
        href="/dogs/perry"
      >
        Perry
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
